torch/testing/_internal/common_utils.py

```diff
@@ -572,6 +577,20 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+# noarch tests are tests that should be only run on one CI configuration,
+# because they don't exercise any interesting platform specific code
+# and so if run once, indicate the test should pass everywhere.
+# See https://github.com/pytorch/pytorch/issues/53743
+def noarchTest(fn):
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        if TEST_SKIP_NOARCH:
+            raise unittest.SkipTest("test is noarch but we are skipping noarch tests due to TEST_SKIP_NOARCH")
```

Comment:
nitty nit (the "but" was a little confusing but it's ok the way it is now too if you want to land)
```suggestion
            raise unittest.SkipTest("test is noarch: we are skipping noarch tests due to TEST_SKIP_NOARCH")
```